### PR TITLE
Added H3IndexFilterOperator

### PR DESCRIPTION
**Pull request by yupeng9**
Updated the H3IndexFilter operator with the correct algorithm. Also fixed the schema and corrupted data in the quickstart example.

Example:
image

### CODE REVIEW

File: pinot-core/src/main/java/org/apache/pinot/core/operator/filter/H3IndexFilterOperator.java [hunk 1]
```diff
@@ -69,30 +76,16 @@ public H3IndexFilterOperator(Predicate predicate, IndexSegment indexSegment, int
     } else {
       throw new RuntimeException("Expecting one of the arguments of ST_DISTANCE to be an identifier");
```
Comment (kishoreg) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/filter/H3IndexFilterOperator.java:77]: better to add this logic in filter plan node and fallback to expressionbased filter operator

File: pinot-core/src/main/java/org/apache/pinot/core/plan/FilterPlanNode.java [hunk 2]
```diff
@@ -93,6 +89,25 @@ public BaseFilterOperator run() {
...
+      if (type != Predicate.Type.RANGE) {
+        return false;
+      }
+      RangePredicate rangePredicate = (RangePredicate) predicate;
+      // TODO: support lower bound
+      if (rangePredicate.getUpperBound().equals(RangePredicate.UNBOUNDED) || !rangePredicate.getLowerBound()
+          .equals(RangePredicate.UNBOUNDED)) {
+        return false;
+      }
+      String columnName = function.getArguments().get(0).getIdentifier();
```
Comment (kishoreg) [inline pinot-core/src/main/java/org/apache/pinot/core/plan/FilterPlanNode.java:104]: add a check that the first argument is identifier

File: pinot-tools/src/main/java/org/apache/pinot/tools/GenericQuickstart.java [hunk 3]
```diff
@@ -106,19 +106,19 @@ public void execute()
 
     Runtime.getRuntime().addShutdownHook(new Thread(() -> {
       try {
-        printStatus(Color.GREEN, "***** Shutting down GitHubEventsQuickStart *****");
+        printStatus(Color.GREEN, "***** Shutting down GenericQuickStart *****");
```
Comment (kishoreg) [inline pinot-tools/src/main/java/org/apache/pinot/tools/GenericQuickstart.java:109]: GenericQuickstart -> <TableName> quickstart cluster or just QuickstartCluster

**Review by kishoreg — APPROVED** [review]

File: pinot-core/src/main/java/org/apache/pinot/core/operator/filter/H3IndexFilterOperator.java (hunk 1 above)
Comment (yupeng9) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/filter/H3IndexFilterOperator.java:77]: Yes, filter plan node has this logic, so this is defensive checking.

**Comment by yupeng9** [comment]
@kishoreg comments addressed. PTAL

**Pull request merged**